lib/shelley/bench/Restore.hs

```diff
@@ -223,78 +224,67 @@ cardanoRestoreBench tr c socketFile = do
+                True -- Write progress to .timelog file
+                (unsafeMkPercentage 1)
+                benchmarksRnd
+
+    let benchRestoreSeqWithOwnership p = do
+            let benchname = showPercentFromPermille p <> "-percent-seq"
+            bench_restoration
+                networkProxy
+                (trMessageText tr)
+                socketFile
+                np
+                vData
+                (T.unpack benchname)
+                [walletSeq benchname
+                    $ mkSeqAnyState' p networkProxy]
+                True -- Write progress to .timelog file
+                (unsafeMkPercentage 1)
+                benchmarksSeq
+
     runBenchmarks
-        [ bench_restoration @_ @ShelleyKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0-percent-seq" $ mkSeqAnyState' @0 networkProxy)
-            benchmarksSeq
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0-percent-rnd" $ mkRndAnyState @0)
-            benchmarksRnd
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0.1-percent-rnd" $ mkRndAnyState @1)
-            benchmarksRnd
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0.2-percent-rnd" $ mkRndAnyState @2)
-            benchmarksRnd
-
-        , bench_restoration @_ @ByronKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletRnd "0.4-percent-rnd" $ mkRndAnyState @4)
-            benchmarksRnd
-
-         , bench_restoration @_ @ShelleyKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0.1-percent-seq" $ mkSeqAnyState' @1 networkProxy)
-            benchmarksSeq
-
-         , bench_restoration @_ @ShelleyKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0.2-percent-seq" $ mkSeqAnyState' @2 networkProxy)
-            benchmarksSeq
-
-         , bench_restoration @_ @ShelleyKey
-            networkProxy
-            (trMessageText tr)
-            socketFile
-            np
-            vData
-            (walletSeq "0.4-percent-seq" $ mkSeqAnyState' @4 networkProxy)
-            benchmarksSeq
+        [ benchRestoreMultipleWallets 1 (unsafeMkPercentage 0.1)
+        , benchRestoreMultipleWallets 10 (unsafeMkPercentage 0.01)
+        , benchRestoreMultipleWallets 100 (unsafeMkPercentage 0.01)
```

Comment:
Why not using the same percentage here? That will skew the benchmark a bit, wouldn't it?

Reply:
> Why not using the same percentage here?

To make it run in a few minutes rather than more than an hour.

See https://github.com/input-output-hk/cardano-wallet/pull/2243#discussion_r505444269

> That will skew the benchmark a bit, wouldn't it?

Well, only if you interpret in the wrong way.

Reply:
I mean, why not: 

```
        [ benchRestoreMultipleWallets 1 (unsafeMkPercentage 0.01)
        , benchRestoreMultipleWallets 10 (unsafeMkPercentage 0.01)
        , benchRestoreMultipleWallets 100 (unsafeMkPercentage 0.01)
```

The first `0.1` looks like a typo to me ?

Reply:
Ah — the extrapolation from 0.01 to 0.1 is 54% off because it is so fast (imprecise start and stop times + ramping-up time, I take).